WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -374,18 +376,31 @@ public static void showAztecEditorReleaseNotes(Activity activity) {
      * open the passed url in the device's external browser
      */
     public static void openUrlExternal(Context context, @NonNull String url) {
+        Uri uri = Uri.parse(url);
+        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+
         try {
             // disable deeplinking activity so to not catch WP URLs
             WPActivityUtils.disableComponent(context, ReaderPostPagerActivity.class);
 
-            Uri uri = Uri.parse(url);
-            Intent intent = new Intent(Intent.ACTION_VIEW, uri);
             context.startActivity(intent);
             AppLockManager.getInstance().setExtendedTimeout();
 
         } catch (ActivityNotFoundException e) {
             ToastUtils.showToast(context, context.getString(R.string.no_default_app_available_to_open_link), ToastUtils.Duration.LONG);
             AppLog.e(AppLog.T.UTILS, "No default app available on the device to open the link: " + url, e);
+        } catch (SecurityException se) {
+            AppLog.e(AppLog.T.UTILS, "Error opening url in default browser. Url: " + url, se);
+
+            List<ResolveInfo> infos = context.getPackageManager().queryIntentActivities(intent, 0);
+            if (infos.size() == 1) {
+                // there's only one handler and apparently it caused the exception so, just inform and bail
+                AppLog.d(AppLog.T.UTILS, "Only one url handler found so, bailing.");
+                ToastUtils.showToast(context, context.getString(R.string.cant_open_url));
+            } else {
+                Intent chooser = Intent.createChooser(intent, context.getString(R.string.error_please_choose_browser));
+                context.startActivity(chooser);
```

Comment:
Niptick: Should we re-set the AppLock extended timeout here? `AppLockManager.getInstance().setExtendedTimeout();`

Reply:
Hah, not a nitpick at all, good catch! Fixed with 16e9370.